### PR TITLE
[Robot Testing]: Added dedicated category, iCubTest app instructions and proto wiki migration

### PR DESCRIPTION
**Pull request by gsisinna**
As per the title, the following PR intends to integrate in the official documentation all those pages and references useful to perform testing on iCub, starting from production tests for encoders and joint limits as well as higher level functional tests like the redball demo.

In particular with https://github.com/icub-tech-iit/documentation/commit/12f423e0dacee817930aa8ce9b94893a0933c4bb has been clarified more about the procedure for starting the new iCubTest application developed for automated, docker deployment-based testing.

### CODE REVIEW

File: docs/icub_testing/icubtest_application.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+Once an option is selected from those available and the START button is pressed, the deployment procedure will also begin on the icub-head and a new yarprobotinterface should launch on the robot (if not already present). Once the suite is finished (e.g. encoders-icub.xml) the results will be saved in the directory indicated in the CUSTOM_PATH field or in the default directory `/tmp/icub-test_results`.
+
+# How to properly stop the application
+In order to properly clean the various nodes in the cluster, it will be necessary to perform the application shutdown in this order:
+
+1. Wait for the end of the tests
+2. Press the STOP button and wait for all nodes in the docker swarm to be cleaned/deleted
+3. Close the application GUI from the X present in the window
+
+**IMPORTANT: For any emergency or safety STOP, always keep the FAULT button handy and press it to switch off power to the motors.**
```
Comment (pattacini) [inline docs/icub_testing/icubtest_application.md:40]: You may use [MkDocs Admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/) here. I think the `warning` type is appropriate.

**Review by pattacini — CHANGES_REQUESTED** [review]
Hi @gsisinna 
Thanks for your nice contribution!

## Duplicates
I've spotted some repetitions I would honestly avoid.

The following topics are already contained in the [iCub First Steps](https://icub-tech-iit.github.io/documentation/icub_starter_kits/first_steps/):
- Face Expressions
- Red Ball (Manual)
- Skin

I'd like to stick to the starter kit; therefore, you may consider integrating therein any further note and/or text you came up with.
Then, remember to clean up the PR removing unnecessary images.

## Images
Also, please, center the images using the syntax below:

```markdown
<center>
<img src ="../img/Picture1.png"/>
</center>
```

## TOC
Somehow, your rendered pages do not show the Table-Of-Contents section on the top-right corner.
I can look into it once the PR is merged.

File: docs/icub_testing/icubtest_application.md (hunk 1 above)
Comment (gsisinna) [reply docs/icub_testing/icubtest_application.md]: Done!

**Review by pattacini — CHANGES_REQUESTED** [review]
Fantastic, thanks @gsisinna 👍🏻 

Could you only please remove the image `versions.PNG`, which doesn't seem to be used?

**Review by pattacini — APPROVED** [review]

**Pull request merged**